### ISSUE TITLE
Odd behavior when using git URLs in pyproject.toml

### ISSUE THREAD

**Issue by fredrikaverpil** [mtkennerly/poetry-dynamic-versioning#21]
I have the following in my pyproject.toml in pkg1:

```
[tool.poetry.dependencies]
python = "^3.7"
pkg2 = { git = "ssh://server/project/repo" }

[tool.poetry-dynamic-versioning]
enable = true
vcs = "git"
pattern  = "^(?P<base>\\d+\\.\\d+\\.\\d+)(-?((?P<stage>[a-zA-Z]+)\\.?(?P<revision>\\d+)?))?"
format-jinja = """
    {%- if distance == 0 -%}
        {{- base -}}
    {%- else -%}
        {{- bump_version(base, 2) }}.dev{{ distance - 1 }}+g{{ commit }}.d{{ datetime.now().strftime('%Y%m%d%H%M%S') -}}
    {%- endif -%}
"""
format-jinja-imports = [
    { module = "datetime", item = "datetime" }
]
```

And during `poetry install -v` I can see that the dependency `pkg2` is being installed as:

```
- Installing pkg2 (0.0.1.dev1+gda05bfa.d20200827155343 6d5b826)
```

The funny thing here is that pk2 does not use poetry-dynamic-versioning. As you can see the hashes also do not match. It also has version git tags (although its pyproject.toml says 0.0.0). Seems like this info is actually taken from pkg1 (rather than from pkg2).

I presume this might be a limitation of poetry-dynamic-versioning?

**Comment by mtkennerly** [comment]
Good find! That's a limitation of the current design, but I'll see if I can make it handle this better.

For reference, does this break anything for you? I can imagine that it would break any code that checks pkg2's version (e.g., enable a feature if it's new enough), but just want to make sure it's not worse than that.

**Comment by fredrikaverpil** [comment]
No it does not appear to break anything, but I haven't tried pointing to a specific commit. I am using the latest commit on master in my case, and this seems to work fine.

**Comment by fredrikaverpil** [comment]
Unfortunately I hit an issue in production which prevents me from using poetry-dynamic-versioning altogether:

```
Because pkg1 (1.2.1.dev0+gd3ce74c.d20200915054707) depends on pkg2 (1.2.1.dev0+gd3ce74c.d20200915054707 git)
 and pkg3 depends on pkg2 (3.1.2.dev0+gf0996b2.d20200914095735 url), pkg1 is forbidden.
So, because pkg3 depends on pkg1 (1.2.1.dev0+gd3ce74c.d20200915054707 git), version solving failed.
```

So, I am pretty sure that this actually embeds the wrong versions inside the metadata of a wheel. The wheel becomes incompatible with its sub-dependencies.

**Comment by mtkennerly** [comment]
Phew 😅 I did start looking into some better handling for this, by the way, but I've been a bit busy and haven't managed to finish it up yet. Hoping to have some news soon.

**Comment by mtkennerly** [comment]
@fredrikaverpil This is now fixed in [v0.9.0 of the plugin](https://github.com/mtkennerly/poetry-dynamic-versioning/releases/tag/v0.9.0). Could you try it out and let me know if you still see any issues with this?

**Comment by fredrikaverpil** [comment]
Yeah this looks great now! ❤🤘
Thank you so much for fixing this!!
